Program.cs

```diff
@@ -1,4 +1,125 @@
-﻿using System;
-Console.WriteLine ("Hello, World!");
+﻿// --------------------------------------------------------------------------------------------
+// Training ~ A training program for interns at Metamation, Batch - July 2023.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------------
+// Program.cs
+// A06: Eight Queens
+// On a standard chessboard 8 queens have to be placed so that no queen can attack any other.
+// As per the rules of chess, the queen can move horizontally, vertically or in a diagonal.
+// No two queens can be on the same row or column or diagonal.
+// --------------------------------------------------------------------------------------------
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
 
-   
\ No newline at end of file
+namespace Training {
+   #region Program ------------------------------------------------------------------------------
+   /// <summary>EightQueens</summary>
+   class Program {
+      #region Methods ---------------------------------------------
+      /// <summary>Prints Eight Queens</summary>
+      /// <param name="args">Arguments</param>
+      static void Main (string[] args) {
+         Console.WriteLine ("Press U for unique solution to display else press any key for all solution: ");
+         bool isUnique = false;
+         if (Console.ReadKey (true).Key == ConsoleKey.U) isUnique = true;
+         QueenPosition (0, isUnique);
+         ChessBoard ();
+      }
+
+      /// <summary>Check the solution is unique and store it</summary>
+      /// <param name="pos">Solution array</param>
+      static void AddUniqueSoln (int[] pos) {
+         var tmp = pos;
+         for (int i = 0; i < 4; i++) {
+            if (Exists (tmp = Rotate (tmp))) return;
+            // Checks for mirror solution
+            if (Exists (tmp.Reverse ().ToArray ())) return;
+         }
+         sFinalSoln.Add (pos);
+      }
+
+      /// <summary>Display the chessboard with eight queens</summary>
+      static void ChessBoard () {
+         Console.OutputEncoding = Encoding.UTF8;
+         for (int i = 1; i <= sFinalSoln.Count; i++) {
+            Console.CursorTop = 2; Console.CursorLeft = 0;
+            Console.WriteLine ($"Solution {i} of {sFinalSoln.Count}");
+            var soln = sFinalSoln[i - 1];
+            for (int j = 0; j <= 8; j++) {

```

Comment:
Can you optimize it better?

The lines of code could further be reduced. The design chars for top and bottom rows are fixed. We need _for loop_ only to print the queens with the design chars for middle.